File: Java/FizzBuzzProject/src/main/java/io/sample/fizzbuzz/FizzBuzzGenerator.java
package io.sample.fizzbuzz;

public class FizzBuzzGenerator {
    public static String generate(int value) {
        if (isDivisibleBy3(value) && isDivisibleBy5(value)) {
            return "fizzbuzz";
        } else if (isDivisibleBy3(value)) {
            return "fizz";
        } else if (isDivisibleBy5(value)) {
            return "buzz";
        }
        return String.valueOf(value);
    }

    private static boolean isDivisibleBy3(int value) {
        return value % 3 == 0;
    }

    private static boolean isDivisibleBy5(int value) {
        return value % 5 == 0;
    }
}

File: csharp/BeingCraftsman.FizzBuzz/FizzBuzzGenerator.cs
namespace BeingCraftsman.FizzBuzz
{
    public static class FizzBuzzGenerator
    {
        public static string Generate(int input)
        {
            if (IsDivisibleBy5(input) && IsDivisibleBy3(input))
            {
                return "FizzBuzz";
            }

            if (IsDivisibleBy3(input))
            {
                return "Fizz";
            }

            if (IsDivisibleBy5(input))
            {
                return "Buzz";
            }

            return input.ToString();
        }

        private static bool IsDivisibleBy3(int input)
        {
            return input % 3 == 0;
        }

        private static bool IsDivisibleBy5(int input)
        {
            return input % 5 == 0;
        }
    }
}
